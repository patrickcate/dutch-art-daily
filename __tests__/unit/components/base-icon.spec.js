import BaseIcon from '@components/base-icon.vue'

describe('BaseIcon Component', () => {
  let wrapper
  let options

  beforeEach(() => {
    options = {
      propsData: {
        icon: {
          viewBox: '0 0 10 10',
          url: 'test.svg',
        },
      },
    }

    wrapper = createWrapper(BaseIcon, options)
  })

  it('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBe(true)
  })

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('renders with custom width and height', () => {
    wrapper = createWrapper(BaseIcon, {
      propsData: {
        ...options.propsData,
        width: '20px',
        height: '20px',
      },
    })
    expect(wrapper).toMatchSnapshot()
  })
})
