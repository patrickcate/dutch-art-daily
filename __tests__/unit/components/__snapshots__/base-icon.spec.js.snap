// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseIcon Component renders correctly 1`] = `
<svg viewBox="0 0 10 10" width="1em" height="1em" class="base-icon">
  <use xlink:href="test.svg"></use>
</svg>
`;

exports[`BaseIcon Component renders with custom width and height 1`] = `
<svg viewBox="0 0 10 10" width="20px" height="20px" class="base-icon">
  <use xlink:href="test.svg"></use>
</svg>
`;
